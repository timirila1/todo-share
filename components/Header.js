import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { colors } from '../services/consts';

export const Header = (props) => {
    const { title } = props;
    return (
        <View style={styles.container}>
            <Text style={styles.title}>{title}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        paddingTop: 64,
        paddingLeft: 40,
        paddingRight: 20,
    },
    title: {
        fontSize: 48,
        fontWeight: 'bold',
        color: colors.main,
    },
});

Header.propTypes = {
    title: PropTypes.string,
};
