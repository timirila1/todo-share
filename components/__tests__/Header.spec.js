import 'react-native';
import React from 'react';
import { Header } from '../Header';
import renderer from 'react-test-renderer';


it('Render Header with title', async () => {
    const tree = renderer.create(<Header title="test"/>).toJSON();
    expect(tree).toMatchSnapshot();
});
