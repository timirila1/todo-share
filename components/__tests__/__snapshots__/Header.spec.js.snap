// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header with title 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "paddingBottom": 40,
      "paddingLeft": 40,
      "paddingRight": 10,
      "paddingTop": 52,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#5e1c73",
        "fontSize": 48,
        "fontWeight": "300",
      }
    }
  >
    test
  </Text>
</View>
`;
